app/src/main/java/io/gnosis/safe/ui/transactions/TransactionListViewModel.kt

```diff
@@ -133,7 +132,14 @@ class TransactionListViewModel
                             localOwners = owners
                         )
                     } else {
-                        TransactionView.Unknown
+                        getTransactionView(
```

Comment:
Under what condition can this branch happen? There is a local transaction with the same nonce, but a different timestamp, thus one of the conflicting transactions that were not selected for execution. Why is the view item created then?

Reply:
This branch is executed for any incoming  history transfer (and safe creation, and maybe even module txs). As they are not AWAITING_EXECUTION an do have `txLocal?.submittedAt == null`

Reply:
txLocal would be null then

Reply:
IIRC txLocal was not null

Reply:
look at line 94:
```kotlin
 if (type == TransactionPagingSource.Type.QUEUE ) {
            txLocal = transactionLocalRepository.updateLocalTxLatest(safe)
        }
```
for history transactions txLocal should be null. Please add explicit else branch 
```kotlin
 if (type == TransactionPagingSource.Type.QUEUE ) {
            txLocal = transactionLocalRepository.updateLocalTxLatest(safe)
        } else {
txLocal = null
}
```
and then the getTransactionView can be replaced with TransactionView.Unknown. Because this branch should not be reachable for historic transactions.